### PR TITLE
Add WaitForDiscoveryService test helper

### PR DESCRIPTION
**Pull request by kevingosse**
## Summary of changes

Add a helper to wait for the DiscoveryService to be initialized, and use it in the Datastream.RabbitMQ sample

## Reason for change

It seems like the sample app was exiting before the service had a chance to initialize, which caused tests to fail.

## Other details

Also Samples.DataStreams.RabbitMQ wasn't in the right folder in the solution tree, so I fixed that.

### CODE REVIEW

File: Datadog.Trace.sln
Comment (andrewlock) [inline Datadog.Trace.sln:1]: Fairly sure you don't need any solution changes 🤨

**Review by andrewlock — APPROVED** [review]
🤞

File: tracer/test/test-applications/integrations/Samples.DataStreams.RabbitMQ/Program.cs [hunk 1]
```diff
@@ -12,8 +13,10 @@ public static class Program
         private static readonly string Message = nameof(Message);
         private static readonly string Host = Environment.GetEnvironmentVariable("RABBITMQ_HOST") ?? "localhost";
         
-        public static void Main(string[] args)
+        public static async Task Main(string[] args)
         {
+            await SampleHelpers.WaitForDiscoveryService();
```
Comment (pierotibou) [inline tracer/test/test-applications/integrations/Samples.DataStreams.RabbitMQ/Program.cs:18]: Should we extend this to other samples or there's a particular reason it's needed for this sample?

**Review by pierotibou — APPROVED** [review]

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (4023)](https://github.com/DataDog/dd-trace-dotnet/tree/ae8d5675c5f29033efebe66eb5174ccf3d65b04a)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4023) - mean (3,032ms)  : 2895, 3168
     .   : milestone, 3032,
    master - mean (3,025ms)  : 2902, 3149
     .   : milestone, 3025,

    section CallTarget+Inlining+NGEN
    This PR (4023) - mean (3,642ms)  : 3569, 3715
     .   : milestone, 3642,
    master - mean (3,635ms)  : 3573, 3697
     .   : milestone, 3635,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4023) - mean (3,135ms)  : 2979, 3292
     .   : milestone, 3135,
    master - mean (3,185ms)  : 3041, 3330
     .   : milestone, 3185,

    section CallTarget+Inlining+NGEN
    This PR (4023) - mean (3,513ms)  : 3426, 3599
     .   : milestone, 3513,
    master - mean (3,516ms)  : 3426, 3606
     .   : milestone, 3516,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4023) - mean (3,089ms)  : 2960, 3218
     .   : milestone, 3089,
    master - mean (3,135ms)  : 2980, 3291
     .   : milestone, 3135,

    section CallTarget+Inlining+NGEN
    This PR (4023) - mean (3,478ms)  : 3359, 3598
     .   : milestone, 3478,
    master - mean (3,482ms)  : 3363, 3601
     .   : milestone, 3482,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4023) - mean (188ms)  : 184, 192
     .   : milestone, 188,
    master - mean (187ms)  : 184, 191
     .   : milestone, 187,

    section CallTarget+Inlining+NGEN
    This PR (4023) - mean (914ms)  : 884, 945
     .   : milestone, 914,
    master - mean (918ms)  : 878, 958
     .   : milestone, 918,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4023) - mean (366ms)  : 362, 370
     .   : milestone, 366,
    master - mean (366ms)  : 360, 372
     .   : milestone, 366,

    section CallTarget+Inlining+NGEN
    This PR (4023) - mean (1,057ms)  : 1023, 1091
     .   : milestone, 1057,
    master - mean (1,056ms)  : 1015, 1097
     .   : milestone, 1056,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4023) - mean (358ms)  : 352, 363
     .   : milestone, 358,
    master - mean (356ms)  : 349, 363
     .   : milestone, 356,

    section CallTarget+Inlining+NGEN
    This PR (4023) - mean (1,009ms)  : 965, 1053
     .   : milestone, 1009,
    master - mean (1,007ms)  : 969, 1045
     .   : milestone, 1007,

```

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #4023 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9): 
* All benchmarks have **the same** speed
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|591μs|170ns|589ns|0| 0|0|2.63&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`WriteAndFlushEnrichedTraces`|net472|817μs|3.77μs|19.6μs|0.401| 0|0|3.22&nbsp;KB|
|#4023|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|585μs|321ns|1.24μs|0| 0|0|2.62&nbsp;KB|
|#4023|`WriteAndFlushEnrichedTraces`|net472|805μs|444ns|1.66μs|0.401| 0|0|3.22&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AppSecBodyBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`AllCycleSimpleBody`|netcoreapp3.1|102μs|535ns|2.67μs|0| 0|0|1.66&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`AllCycleSimpleBody`|net472|94μs|273ns|1.06μs|0.238| 0|0|1.71&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`AllCycleMoreComplexBody`|netcoreapp3.1|271μs|1.32μs|7.72μs|0| 0|0|9.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`AllCycleMoreComplexBody`|net472|272μs|943ns|3.53μs|1.47| 0|0|9.31&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`ObjectExtractorSimpleBody`|netcoreapp3.1|180ns|0.05ns|0.18ns|0.00371| 0|0|272&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`ObjectExtractorSimpleBody`|net472|144ns|0.21ns|0.786ns|0.0446| 0|0|281&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`ObjectExtractorMoreComplexBody`|netcoreapp3.1|4.02μs|2.65ns|9.93ns|0.0499| 0|0|3.78&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`ObjectExtractorMoreComplexBody`|net472|3.91μs|2.94ns|10.6ns|0.618| 0.00576|0|3.89&nbsp;KB|
|#4023|`AllCycleSimpleBody`|netcoreapp3.1|99.3μs|577ns|4.93μs|0| 0|0|1.66&nbsp;KB|
|#4023|`AllCycleSimpleBody`|net472|94μs|145ns|542ns|0.23| 0|0|1.71&nbsp;KB|
|#4023|`AllCycleMoreComplexBody`|netcoreapp3.1|270μs|1.43μs|7.44μs|0| 0|0|9.14&nbsp;KB|
|#4023|`AllCycleMoreComplexBody`|net472|271μs|1.36μs|6.1μs|1.45| 0|0|9.31&nbsp;KB|
|#4023|`ObjectExtractorSimpleBody`|netcoreapp3.1|177ns|0.209ns|0.781ns|0.00371| 0|0|272&nbsp;B|
|#4023|`ObjectExtractorSimpleBody`|net472|144ns|0.173ns|0.669ns|0.0446| 0|0|281&nbsp;B|
|#4023|`ObjectExtractorMoreComplexBody`|netcoreapp3.1|4.11μs|5.15ns|19.9ns|0.0512| 0|0|3.78&nbsp;KB|
|#4023|`ObjectExtractorMoreComplexBody`|net472|3.93μs|7.75ns|30ns|0.617| 0.00585|0|3.89&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`SendRequest`|netcoreapp3.1|183μs|219ns|850ns|0.182| 0|0|20.36&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`SendRequest`|net472|0.000687ns|0.000393ns|0.00147ns|0| 0|0|0&nbsp;b|
|#4023|`SendRequest`|netcoreapp3.1|179μs|246ns|953ns|0.265| 0|0|20.36&nbsp;KB|
|#4023|`SendRequest`|net472|0.000461ns|0.000208ns|0.000751ns|0| 0|0|0&nbsp;b|
</details>


<details><summary><strong>Benchmarks.Trace.CIVisibilityProtocolWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|606μs|2.02μs|7.82μs|0.302| 0|0|41.76&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`WriteAndFlushEnrichedTraces`|net472|787μs|1.75μs|6.53μs|8.31| 2.37|0.396|53.24&nbsp;KB|
|#4023|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|609μs|1.02μs|3.93μs|0.3| 0|0|41.69&nbsp;KB|
|#4023|`WriteAndFlushEnrichedTraces`|net472|765μs|2.26μs|8.44μs|8.15| 2.33|0.388|53.22&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`ExecuteNonQuery`|netcoreapp3.1|1.19μs|0.428ns|1.54ns|0.012| 0|0|904&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`ExecuteNonQuery`|net472|1.7μs|1.61ns|6.23ns|0.144| 0.000854|0|907&nbsp;B|
|#4023|`ExecuteNonQuery`|netcoreapp3.1|1.26μs|0.793ns|2.97ns|0.012| 0|0|904&nbsp;B|
|#4023|`ExecuteNonQuery`|net472|1.7μs|2.04ns|7.91ns|0.144| 0|0|907&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`CallElasticsearch`|netcoreapp3.1|1.34μs|0.435ns|1.69ns|0.0146| 0|0|1.06&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`CallElasticsearch`|net472|2.19μs|0.805ns|3.12ns|0.176| 0.00109|0|1.11&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`CallElasticsearchAsync`|netcoreapp3.1|1.42μs|0.503ns|1.95ns|0.0164| 0|0|1.18&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`CallElasticsearchAsync`|net472|2.29μs|0.51ns|1.98ns|0.197| 0.00115|0|1.24&nbsp;KB|
|#4023|`CallElasticsearch`|netcoreapp3.1|1.35μs|0.678ns|2.54ns|0.0141| 0|0|1.06&nbsp;KB|
|#4023|`CallElasticsearch`|net472|2.23μs|0.912ns|3.53ns|0.176| 0.00111|0|1.11&nbsp;KB|
|#4023|`CallElasticsearchAsync`|netcoreapp3.1|1.39μs|0.441ns|1.65ns|0.016| 0|0|1.18&nbsp;KB|
|#4023|`CallElasticsearchAsync`|net472|2.3μs|1.35ns|5.23ns|0.197| 0.00115|0|1.24&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`ExecuteAsync`|netcoreapp3.1|1.48μs|0.582ns|2.18ns|0.0171| 0|0|1.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`ExecuteAsync`|net472|1.78μs|0.453ns|1.75ns|0.206| 0.000892|0|1.3&nbsp;KB|
|#4023|`ExecuteAsync`|netcoreapp3.1|1.45μs|1.09ns|4.21ns|0.0169| 0|0|1.28&nbsp;KB|
|#4023|`ExecuteAsync`|net472|1.91μs|0.745ns|2.88ns|0.206| 0|0|1.3&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`SendAsync`|netcoreapp3.1|4.32μs|1.48ns|5.34ns|0.0366| 0|0|2.66&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`SendAsync`|net472|6.8μs|1.6ns|5.97ns|0.481| 0|0|3.03&nbsp;KB|
|#4023|`SendAsync`|netcoreapp3.1|4.22μs|0.974ns|3.37ns|0.036| 0|0|2.66&nbsp;KB|
|#4023|`SendAsync`|net472|6.86μs|2.14ns|8.29ns|0.48| 0|0|3.03&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`EnrichedLog`|netcoreapp3.1|1.95μs|0.721ns|2.5ns|0.0253| 0|0|1.83&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`EnrichedLog`|net472|2.59μs|3.58ns|13.9ns|0.283| 0|0|1.79&nbsp;KB|
|#4023|`EnrichedLog`|netcoreapp3.1|1.94μs|1.02ns|3.82ns|0.0245| 0|0|1.83&nbsp;KB|
|#4023|`EnrichedLog`|net472|2.62μs|2.22ns|8.58ns|0.284| 0|0|1.79&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`EnrichedLog`|netcoreapp3.1|116μs|109ns|376ns|0| 0|0|4.42&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`EnrichedLog`|net472|147μs|88ns|341ns|0.733| 0.22|0|4.63&nbsp;KB|
|#4023|`EnrichedLog`|netcoreapp3.1|117μs|116ns|451ns|0.0586| 0|0|4.42&nbsp;KB|
|#4023|`EnrichedLog`|net472|147μs|114ns|440ns|0.731| 0.219|0|4.63&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`EnrichedLog`|netcoreapp3.1|3.86μs|1.62ns|6.06ns|0.0538| 0|0|3.9&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`EnrichedLog`|net472|5.13μs|1.28ns|4.97ns|0.564| 0.00255|0|3.56&nbsp;KB|
|#4023|`EnrichedLog`|netcoreapp3.1|3.84μs|1.95ns|7.29ns|0.0517| 0|0|3.9&nbsp;KB|
|#4023|`EnrichedLog`|net472|5.1μs|1.16ns|4.48ns|0.565| 0.00255|0|3.56&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`SendReceive`|netcoreapp3.1|1.56μs|0.808ns|3.02ns|0.0172| 0|0|1.3&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`SendReceive`|net472|2.01μs|0.985ns|3.68ns|0.212| 0.001|0|1.34&nbsp;KB|
|#4023|`SendReceive`|netcoreapp3.1|1.5μs|0.59ns|2.21ns|0.0172| 0|0|1.3&nbsp;KB|
|#4023|`SendReceive`|net472|1.85μs|2.19ns|8.21ns|0.213| 0.000914|0|1.34&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`EnrichedLog`|netcoreapp3.1|3.71μs|1.41ns|5.07ns|0.0243| 0|0|1.78&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`EnrichedLog`|net472|4.35μs|1.08ns|3.88ns|0.35| 0|0|2.21&nbsp;KB|
|#4023|`EnrichedLog`|netcoreapp3.1|3.8μs|1.08ns|4.05ns|0.0247| 0|0|1.78&nbsp;KB|
|#4023|`EnrichedLog`|net472|4.39μs|1.15ns|4.3ns|0.35| 0|0|2.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`StartFinishSpan`|netcoreapp3.1|666ns|0.441ns|1.65ns|0.00969| 0|0|720&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`StartFinishSpan`|net472|914ns|0.774ns|3ns|0.121| 0|0|762&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`StartFinishScope`|netcoreapp3.1|796ns|0.505ns|1.95ns|0.0113| 0|0|840&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`StartFinishScope`|net472|1.18μs|0.575ns|2.15ns|0.133| 0|0|842&nbsp;B|
|#4023|`StartFinishSpan`|netcoreapp3.1|716ns|0.747ns|2.79ns|0.00972| 0|0|720&nbsp;B|
|#4023|`StartFinishSpan`|net472|928ns|0.211ns|0.788ns|0.121| 0|0|762&nbsp;B|
|#4023|`StartFinishScope`|netcoreapp3.1|785ns|0.299ns|1.08ns|0.0117| 0|0|840&nbsp;B|
|#4023|`StartFinishScope`|net472|1.07μs|0.306ns|1.14ns|0.134| 0|0|842&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`RunOnMethodBegin`|netcoreapp3.1|903ns|0.538ns|1.86ns|0.0114| 0|0|840&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/31ea9df13a5a9dccb3c49625fa4b0cc2201201b9)|`RunOnMethodBegin`|net472|1.23μs|1.22ns|4.73ns|0.134| 0|0|842&nbsp;B|
|#4023|`RunOnMethodBegin`|netcoreapp3.1|961ns|0.322ns|1.25ns|0.011| 0|0|840&nbsp;B|
|#4023|`RunOnMethodBegin`|net472|1.23μs|0.722ns|2.79ns|0.133| 0|0|842&nbsp;B|
</details>

**Pull request merged**